### PR TITLE
Entity cache: influence the cache key through a context entry

### LINKED ISSUE #4190: Entity caching: use a context key as part of the cache key
we need a way to influence the cache key depending on information outside of the query. For the query plan cache, this was done through an entry in the context, we can reuse the same approach here

### PR DESCRIPTION
**Pull request by Geal**
Fix #4190

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: apollo-router/src/plugins/cache/entity.rs [hunk 1]
```diff
@@ -34,6 +34,7 @@ use crate::Context;
 
 const ENTITIES: &str = "_entities";
 pub(crate) const REPRESENTATIONS: &str = "representations";
+pub(crate) const CONTEXT_CACHE_KEY: &str = "entity_cache_key";
```
Comment (bnjjj) [inline apollo-router/src/plugins/cache/entity.rs:37]: Can we namespace it with `apollo.cache::entity_cache_key` for example ?
```suggestion
pub(crate) const CONTEXT_CACHE_KEY: &str = "apollo.cache::entity_cache_key";
```

Comment (Geal) [reply apollo-router/src/plugins/cache/entity.rs:37]: do we use such a convention elsewhere?

Comment (bnjjj) [reply apollo-router/src/plugins/cache/entity.rs:37]: Ok I'm wrong, if it's an exposed key for the user don't namespace it. But put it in public and also provide it in rhai globals please. I did the same for subscription context key

**Review by bnjjj — APPROVED** [review]
Don't forget to document it in #4195

**Pull request merged**